### DISCUSSION TITLE
I want to filter the reported power draw from Zigbee-devices, do I need to make a quirk to do this?

### DISCUSSION THREAD

**Discussion by skasti** [zigpy/zha-device-handlers#3082] [Q&A] [answered]
I have some Namron plug-in relays that act as switches and report power draw and total consumption.
These work fine out of the box, and integrates nicely with HA energy dashboard, but when switching a large load, they seem to intermittently report a power draw of 25100W, which I assume is some sort of max value internally in the device, but obviously not a real value, as the max power draw of the device plugged in is about 2000W, and the relays max rated power is 3600W.

I would like to somehow avoid reporting of these values, so that I can avoid needing to filter them everywhere I use them. Do I need a quirk for this, or is there a simpler way I just don't know about?
I have previously made a custom quirk to get the power-reporting of my Futurehome (tuya) thermostats working, but then I could copy existing quirks to get the base functionality, while with these namron-relays, everything already works, I just want to modify a small part of how they report values, and I don't really know where to start to get a quirk that does not change anything about a device? 😅 

IMHO the manufacturer should have done some more work to avoid this issue in the first place, but here we are.

**Comment by Hedda** [comment, marked as answer]
> with these namron-relays, everything already works, **I just want to modify a small part of how they report values, and I don't really know where to start to get a quirk that does not change anything about a device?** 😅

I understand that should or will be possible with the new "Quirks V2" (ZHA Device Handlers version 2) architectural design, more info here -> https://community.home-assistant.io/t/zha-device-handlers-version-2-zha-quirks-v2-architectural-design-developer-discussion/666460

FYI, ZHA/zigpy developers have introduced initial support for what they call ”Quirks V2” (or Z"HA-Quirks V2") and dmulcahey has begun documentation for easier quirk development in zha-device-handlers (zha-quirks) using the ”Quirks V2” specification here -> https://github.com/zigpy/zha-device-handlers/pull/3019

> These work fine out of the box, and integrates nicely with HA energy dashboard, but when switching a large load, they seem to intermittently report a power draw of 25100W, which I assume is some sort of max value internally in the device, but obviously not a real value, as the max power draw of the device plugged in is about 2000W, and the relays max rated power is 3600W.
> 
> I would like to somehow avoid reporting of these values, so that I can avoid needing to filter them everywhere I use them. Do I need a quirk for this, or is there a simpler way I just don't know about?

Yes you probably have to change multiplier and/or divider in the respective ZHA Device Handler quirk in this “ZHA Device Handlers" (zha-quirks) repository, so regardless if you go with the "Quirks V1" or new "Quirks V2" specification, do open a new "Device support request" issue with device signature + diagnostic information if you can not find an already existing issue for your specific device signature. See -> https://github.com/zigpy/zha-device-handlers/issues and if can not find existing issues for the same device signasure then open a "device support request" as a new issue -> https://github.com/zigpy/zha-device-handlers/issues/new/choose 

PS: Sound like you are already aware of this but FYI, the usual reason why quirks are needed is normally due to firmware reporting non-standard values, see -> https://www.home-assistant.io/integrations/zha#how-to-add-support-for-new-and-unsupported-devices
